src/librustdoc/doctree.rs

```diff
@@ -4,17 +4,47 @@ use rustc_span::{self, Span, Symbol};
 
 use rustc_hir as hir;
 
+/// A wrapper around a [`hir::Item`].
+#[derive(Debug)]
+crate struct Item<'hir> {
+    /// the wrapped item
+    crate hir_item: &'hir hir::Item<'hir>,
+    /// the explicit renamed name
+    crate renamed_name: Option<Symbol>,
+    /// whether the item is from a glob import
+    /// if `from_glob` is true and we see another item with same name,
+    /// then this item can be replaced with that one
+    crate from_glob: bool,
+}
+
+impl<'hir> Item<'hir> {
+    pub(crate) fn new(
+        hir_item: &'hir hir::Item<'hir>,
+        renamed_name: Option<Symbol>,
+        from_glob: bool,
+    ) -> Self {
+        Self { hir_item, renamed_name, from_glob }
+    }
+
+    pub(crate) fn name(&'hir self) -> &'hir Symbol {
+        self.renamed_name.as_ref().unwrap_or(&self.hir_item.ident.name)
+    }
```

Comment:
Symbol is Copy, you can just return it directly.
```suggestion
    pub(crate) fn name(&self) -> Symbol {
        self.renamed_name.unwrap_or(self.hir_item.ident.name)
    }
```